node/src/blockchain/blockchain_bridge.rs

```diff
@@ -175,25 +174,53 @@ impl BlockchainBridge {
         }
     }
 
-    fn handle_report_accounts_payable(
+    fn handle_report_accounts_payable(&self, creditors_msg: ReportAccountsPayable) {
+        let processed_payments = self.handle_report_accounts_payable_inner(creditors_msg);
+        match processed_payments {
+            Ok(payments) => {
+                self.sent_payments_subs_opt
+                    .as_ref()
+                    .expect("Accountant is unbound")
+                    .try_send(SentPayments { payments })
+                    .expect("Accountant is dead");
+            }
+            Err(e) => warning!(self.logger, "{}", e),
+        }
+    }
+
+    fn handle_report_accounts_payable_inner(
         &self,
         creditors_msg: ReportAccountsPayable,
-    ) -> MessageResult<ReportAccountsPayable> {
-        MessageResult(match self.consuming_wallet_opt.as_ref() {
-            Some(consuming_wallet) => {
-                let gas_price = match self.persistent_config.gas_price() {
-                    Ok(num) => num,
-                    Err(err) => {
-                        return MessageResult(Err(format!(
-                            "ReportAccountPayable: gas-price: {:?}",
-                            err
-                        )))
-                    }
-                };
-                Ok(self.process_payments(creditors_msg, gas_price, consuming_wallet))
-            }
+    ) -> Result<Vec<BlockchainResult<Payment>>, String> {
+        match self.consuming_wallet_opt.as_ref() {
+            Some(consuming_wallet) => match self.persistent_config.gas_price() {
+                Ok(gas_price) => {
+                    Ok(self.process_payments(creditors_msg, gas_price, consuming_wallet))
+                }
+                Err(err) => Err(format!("ReportAccountPayable: gas-price: {:?}", err)),
+            },
             None => Err(String::from("No consuming wallet specified")),
-        })
+        }
+    }
+
+    fn handle_retrieve_transactions(&self, msg: RetrieveTransactions) {
+        let retrieved_transactions = self.handle_retrieve_transactions_inner(msg);
+        match retrieved_transactions {
+            Ok(transactions) => self
+                .received_payments_subs_opt
+                .as_ref()
+                .expect("Accountant is unbound")
+                .try_send(ReceivedPayments {
+                    payments: transactions,
+                })
+                .expect("Accountant is dead."),
+            Err(e) => warning!(self.logger, "{}", e),
+        }
+    }
+
+    fn handle_retrieve_transactions_inner(&self, msg: RetrieveTransactions) -> Transactions {
```

Comment:
You're following the pattern given by me, I get it, but not always can you. It doesn't fit here because the 'inner' is very simple and it doesn't deserve its own function, at the end of the day, this is just one function call wrapped into another one and that always smells fishy unless you really want to construct a wrapper (which is a class on its own).

I did that just so I could spare some space in tests, writing less complicated tests, tailored just for the inner's code.

In fact, I think I evaluated it badly I could just do without this finesse. The tests would have probably looked pretty the same, with small differences, maybe. 

So my wish here is that you should get rid of my "something_inner" and your "somethingelse_inner" and reset the tests to address this situation.